### PR TITLE
dnf sbom: handle arch absence

### PR DESCRIPTION
**Pull request by Habbie**
### Short description
fixes the breakage introduced in #14691

### Checklist

I have:
- [x] read the [CONTRIBUTING.md](https://github.com/PowerDNS/pdns/blob/master/CONTRIBUTING.md) document
- [x] compiled this code
- [x] tested this code
- [ ] included documentation (including possible behaviour changes)
- [ ] documented the code
- [ ] added or modified regression test(s)
- [ ] added or modified unit test(s)
- [ ]  checked that this code was merged to master

### CODE REVIEW

**Comment by Habbie** [comment]
non-release dnsdist (because auth builds don't use the extra json args) test build at https://github.com/Habbie/pdns/actions/runs/10963274205

**Comment by Habbie** [comment]
fake dnsdist release build https://github.com/Habbie/pdns/actions/runs/10964295787

**Comment by Habbie** [comment]
<details>
<summary>fake release build</summary>

```json
{
  "bomFormat": "CycloneDX",
  "specVersion": "1.5",
  "version": 1,
  "serialNumber": "urn:uuid:20d7fe22-dc29-47ea-aa7b-17dda149f8e0",
  "metadata": {
    "timestamp": "2024-09-20T18:35:51.449607+00:00",
    "authors": [
      {
        "name": "PowerDNS.COM BV"
      }
    ],
    "component": {
      "name": "dnsdist",
      "bom-ref": "pkg:dnsdist",
      "type": "application",
      "version": "0.0.7-1pdns.el8.x86_64",
      "supplier": {
        "name": "PowerDNS.COM BV",
        "url": [
          "https://www.powerdns.com"
        ]
      },
      "licenses": [
        {
          "license": {
            "id": "GPL-2.0-only"
          }
        }
      ]
    }
  },
  "components": [
    {
      "name": "shadow-utils",
      "bom-ref": "lib:shadow-utils",
      "type": "library",
      "version": "2:4.6-22.el8.x86_64",
      "supplier": {
        "name": "Oracle America"
      },
      "licenses": [
        {
          "license": {
            "name": "BSD and GPLv2+"
          }
        }
      ]
    },
    {
      "name": "systemd",
      "bom-ref": "lib:systemd",
      "type": "library",
      "version": "239-82.0.2.el8.x86_64",
      "supplier": {
        "name": "Oracle America"
      },
      "licenses": [
        {
          "license": {
            "name": "LGPLv2+ and MIT and GPLv2+"
          }
        }
      ]
    },
    {
      "name": "libbpf",
      "bom-ref": "lib:libbpf",
      "type": "library",
      "version": "0.5.0-1.el8.x86_64",
      "supplier": {
        "name": "Oracle America"
      },
      "licenses": [
        {
          "license": {
            "name": "LGPLv2 or BSD"
          }
        }
      ]
    },
    {
      "name": "glibc",
      "bom-ref": "lib:glibc",
      "type": "library",
      "version": "2.28-251.0.2.el8_10.2.x86_64",
      "supplier": {
        "name": "Oracle America"
      },
      "licenses": [
        {
          "license": {
            "name": "LGPLv2+ and LGPLv2+ with exceptions and GPLv2+ and GPLv2+ with exceptions and BSD and Inner-Net and ISC and Public Domain and GFDL"
          }
        }
      ]
    },
    {
      "name": "libcap",
      "bom-ref": "lib:libcap",
      "type": "library",
      "version": "2.48-6.el8_9.x86_64",
      "supplier": {
        "name": "Oracle America"
      },
      "licenses": [
        {
          "license": {
            "name": "BSD or GPLv2"
          }
        }
      ]
    },
    {
      "name": "tinycdb",
      "bom-ref": "lib:tinycdb",
      "type": "library",
      "version": "0.78-9.el8.x86_64",
      "supplier": {
        "name": "Oracle America"
      },
      "licenses": [
        {
          "license": {
            "name": "Public Domain"
          }
        }
      ]
    },
    {
      "name": "openssl-libs",
      "bom-ref": "lib:openssl-libs",
      "type": "library",
      "version": "1:1.1.1k-12.el8_9.x86_64",
      "supplier": {
        "name": "Oracle America"
      },
      "licenses": [
        {
          "license": {
            "name": "OpenSSL and ASL 2.0"
          }
        }
      ]
    },
    {
      "name": "libedit",
      "bom-ref": "lib:libedit",
      "type": "library",
      "version": "3.1-23.20170329cvs.el8.x86_64",
      "supplier": {
        "name": "Oracle America"
      },
      "licenses": [
        {
          "license": {
            "id": "BSD-3-Clause"
          }
        }
      ]
    },
    {
      "name": "fstrm",
      "bom-ref": "lib:fstrm",
      "type": "library",
      "version": "0.6.1-3.el8.x86_64",
      "supplier": {
        "name": "Oracle America"
      },
      "licenses": [
        {
          "license": {
            "id": "MIT"
          }
        }
      ]
    },
    {
      "name": "libgcc",
      "bom-ref": "lib:libgcc",
      "type": "library",
      "version": "8.5.0-22.0.1.el8_10.x86_64",
      "supplier": {
        "name": "Oracle America"
      },
      "licenses": [
        {
          "license": {
            "name": "GPLv3+ and GPLv3+ with exceptions and GPLv2+ with exceptions and LGPLv2+ and BSD"
          }
        }
      ]
    },
    {
      "name": "gnutls",
      "bom-ref": "lib:gnutls",
      "type": "library",
      "version": "3.6.16-8.el8_9.3.x86_64",
      "supplier": {
        "name": "Oracle America"
      },
      "licenses": [
        {
          "license": {
            "name": "GPLv3+ and LGPLv2+"
          }
        }
      ]
    },
    {
      "name": "lmdb-libs",
      "bom-ref": "lib:lmdb-libs",
      "type": "library",
      "version": "0.9.24-2.el8.x86_64",
      "supplier": {
        "name": "Oracle America"
      },
      "licenses": [
        {
          "license": {
            "id": "OLDAP-2.8"
          }
        }
      ]
    },
    {
      "name": "luajit",
      "bom-ref": "lib:luajit",
      "type": "library",
      "version": "2.1.0-0.16beta3.el8.x86_64",
      "supplier": {
        "name": "Fedora Project"
      },
      "licenses": [
        {
          "license": {
            "id": "MIT"
          }
        }
      ]
    },
    {
      "name": "net-snmp-libs",
      "bom-ref": "lib:net-snmp-libs",
      "type": "library",
      "version": "1:5.8-30.0.1.el8.x86_64",
      "supplier": {
        "name": "Oracle America"
      },
      "licenses": [
        {
          "license": {
            "id": "BSD-3-Clause"
          }
        }
      ]
    },
    {
      "name": "net-snmp-agent-libs",
      "bom-ref": "lib:net-snmp-agent-libs",
      "type": "library",
      "version": "1:5.8-30.0.1.el8.x86_64",
      "supplier": {
        "name": "Oracle America"
      },
      "licenses": [
        {
          "license": {
            "id": "BSD-3-Clause"
          }
        }
      ]
    },
    {
      "name": "libnghttp2",
      "bom-ref": "lib:libnghttp2",
      "type": "library",
      "version": "1.33.0-6.el8_10.1.x86_64",
      "supplier": {
        "name": "Oracle America"
      },
      "licenses": [
        {
          "license": {
            "id": "MIT"
          }
        }
      ]
    },
    {
      "name": "re2",
      "bom-ref": "lib:re2",
      "type": "library",
      "version": "20190801-17.el8.x86_64",
      "supplier": {
        "name": "Fedora Project"
      },
      "licenses": [
        {
          "license": {
            "name": "BSD-3-Clause"
          }
        }
      ]
    },
    {
      "name": "libsodium",
      "bom-ref": "lib:libsodium",
      "type": "library",
      "version": "1.0.18-2.el8.x86_64",
      "supplier": {
        "name": "Fedora Project"
      },
      "licenses": [
        {
          "license": {
            "name": "ISC"
          }
        }
      ]
    },
    {
      "name": "libstdc++",
      "bom-ref": "lib:libstdc++",
      "type": "library",
      "version": "8.5.0-22.0.1.el8_10.x86_64",
      "supplier": {
        "name": "Oracle America"
      },
      "licenses": [
        {
          "license": {
            "name": "GPLv3+ and GPLv3+ with exceptions and GPLv2+ with exceptions and LGPLv2+ and BSD"
          }
        }
      ]
    },
    {
      "name": "systemd-libs",
      "bom-ref": "lib:systemd-libs",
      "type": "library",
      "version": "239-82.0.2.el8.x86_64",
      "supplier": {
        "name": "Oracle America"
      },
      "licenses": [
        {
          "license": {
            "name": "LGPLv2+ and MIT"
          }
        }
      ]
    },
    {
      "name": "libxdp",
      "bom-ref": "lib:libxdp",
      "type": "library",
      "version": "1.2.10-1.el8.x86_64",
      "supplier": {
        "name": "Oracle America"
      },
      "licenses": [
        {
          "license": {
            "id": "GPL-2.0-only"
          }
        }
      ]
    },
    {
      "name": "rust",
      "bom-ref": "lib:rust",
      "type": "library",
      "version": "1.78.0",
      "publisher": "https://www.rust-lang.org/",
      "licenses": [
        {
          "license": {
            "id": "MIT"
          }
        }
      ],
      "hashes": [
        {
          "alg": "SHA-256",
          "content": "1377999f189d328ec183676c0e69f21af16e450f8d67f10d1f6cf746951a1d64"
        }
      ]
    },
    {
      "name": "quiche",
      "bom-ref": "lib:quiche",
      "type": "library",
      "version": "0.22.0",
      "publisher": "https://github.com/cloudflare/quiche",
      "licenses": [
        {
          "license": {
            "name": "BSD-2-Clause"
          }
        }
      ],
      "hashes": [
        {
          "alg": "SHA-256",
          "content": "0af8744b07038ee4af8cdb94dd4c11f1a730001944a0ef2f3f03e63715b15268"
        }
      ]
    },
    {
      "name": "h2o",
      "bom-ref": "lib:h2o",
      "type": "library",
      "version": "2.2.6-pdns2",
      "publisher": "https://github.com/h2o/h2o",
      "licenses": [
        {
          "license": {
            "id": "MIT"
          }
        }
      ],
      "hashes": [
        {
          "alg": "SHA-256",
          "content": "e25959c3f9a102e7a332ca0bb8b3f533eb14919d5a60ca999730c2ebee4b548f"
        }
      ]
    }
  ],
  "dependencies": [
    {
      "ref": "pkg:dnsdist",
      "dependsOn": [
        "lib:shadow-utils",
        "lib:systemd",
        "lib:libbpf",
        "lib:glibc",
        "lib:libcap",
        "lib:tinycdb",
        "lib:openssl-libs",
        "lib:libedit",
        "lib:fstrm",
        "lib:libgcc",
        "lib:gnutls",
        "lib:lmdb-libs",
        "lib:luajit",
        "lib:net-snmp-libs",
        "lib:net-snmp-agent-libs",
        "lib:libnghttp2",
        "lib:re2",
        "lib:libsodium",
        "lib:libstdc++",
        "lib:systemd-libs",
        "lib:libxdp",
        "lib:rust",
        "lib:quiche",
        "lib:h2o"
      ]
    }
  ]
}
```

</details>

<details>
<summary>non-release build</summary>

```json
{
  "bomFormat": "CycloneDX",
  "specVersion": "1.5",
  "version": 1,
  "serialNumber": "urn:uuid:7feabc2c-63d7-46ad-964b-c7f00c5ef196",
  "metadata": {
    "timestamp": "2024-09-20T17:11:01.689804+00:00",
    "authors": [
      {
        "name": "PowerDNS.COM BV"
      }
    ],
    "component": {
      "name": "dnsdist",
      "bom-ref": "pkg:dnsdist",
      "type": "application",
      "version": "1.9.0-0.alpha0.dnfsbomversionarchoptional.3899.g857abddfc.1pdns.el8.x86_64",
      "supplier": {
        "name": "PowerDNS.COM BV",
        "url": [
          "https://www.powerdns.com"
        ]
      },
      "licenses": [
        {
          "license": {
            "id": "GPL-2.0-only"
          }
        }
      ]
    }
  },
  "components": [
    {
      "name": "shadow-utils",
      "bom-ref": "lib:shadow-utils",
      "type": "library",
      "version": "2:4.6-22.el8.x86_64",
      "supplier": {
        "name": "Oracle America"
      },
      "licenses": [
        {
          "license": {
            "name": "BSD and GPLv2+"
          }
        }
      ]
    },
    {
      "name": "systemd",
      "bom-ref": "lib:systemd",
      "type": "library",
      "version": "239-82.0.2.el8.x86_64",
      "supplier": {
        "name": "Oracle America"
      },
      "licenses": [
        {
          "license": {
            "name": "LGPLv2+ and MIT and GPLv2+"
          }
        }
      ]
    },
    {
      "name": "libbpf",
      "bom-ref": "lib:libbpf",
      "type": "library",
      "version": "0.5.0-1.el8.x86_64",
      "supplier": {
        "name": "Oracle America"
      },
      "licenses": [
        {
          "license": {
            "name": "LGPLv2 or BSD"
          }
        }
      ]
    },
    {
      "name": "glibc",
      "bom-ref": "lib:glibc",
      "type": "library",
      "version": "2.28-251.0.2.el8_10.2.x86_64",
      "supplier": {
        "name": "Oracle America"
      },
      "licenses": [
        {
          "license": {
            "name": "LGPLv2+ and LGPLv2+ with exceptions and GPLv2+ and GPLv2+ with exceptions and BSD and Inner-Net and ISC and Public Domain and GFDL"
          }
        }
      ]
    },
    {
      "name": "libcap",
      "bom-ref": "lib:libcap",
      "type": "library",
      "version": "2.48-6.el8_9.x86_64",
      "supplier": {
        "name": "Oracle America"
      },
      "licenses": [
        {
          "license": {
            "name": "BSD or GPLv2"
          }
        }
      ]
    },
    {
      "name": "tinycdb",
      "bom-ref": "lib:tinycdb",
      "type": "library",
      "version": "0.78-9.el8.x86_64",
      "supplier": {
        "name": "Oracle America"
      },
      "licenses": [
        {
          "license": {
            "name": "Public Domain"
          }
        }
      ]
    },
    {
      "name": "openssl-libs",
      "bom-ref": "lib:openssl-libs",
      "type": "library",
      "version": "1:1.1.1k-12.el8_9.x86_64",
      "supplier": {
        "name": "Oracle America"
      },
      "licenses": [
        {
          "license": {
            "name": "OpenSSL and ASL 2.0"
          }
        }
      ]
    },
    {
      "name": "libedit",
      "bom-ref": "lib:libedit",
      "type": "library",
      "version": "3.1-23.20170329cvs.el8.x86_64",
      "supplier": {
        "name": "Oracle America"
      },
      "licenses": [
        {
          "license": {
            "id": "BSD-3-Clause"
          }
        }
      ]
    },
    {
      "name": "fstrm",
      "bom-ref": "lib:fstrm",
      "type": "library",
      "version": "0.6.1-3.el8.x86_64",
      "supplier": {
        "name": "Oracle America"
      },
      "licenses": [
        {
          "license": {
            "id": "MIT"
          }
        }
      ]
    },
    {
      "name": "libgcc",
      "bom-ref": "lib:libgcc",
      "type": "library",
      "version": "8.5.0-22.0.1.el8_10.x86_64",
      "supplier": {
        "name": "Oracle America"
      },
      "licenses": [
        {
          "license": {
            "name": "GPLv3+ and GPLv3+ with exceptions and GPLv2+ with exceptions and LGPLv2+ and BSD"
          }
        }
      ]
    },
    {
      "name": "gnutls",
      "bom-ref": "lib:gnutls",
      "type": "library",
      "version": "3.6.16-8.el8_9.3.x86_64",
      "supplier": {
        "name": "Oracle America"
      },
      "licenses": [
        {
          "license": {
            "name": "GPLv3+ and LGPLv2+"
          }
        }
      ]
    },
    {
      "name": "lmdb-libs",
      "bom-ref": "lib:lmdb-libs",
      "type": "library",
      "version": "0.9.24-2.el8.x86_64",
      "supplier": {
        "name": "Oracle America"
      },
      "licenses": [
        {
          "license": {
            "id": "OLDAP-2.8"
          }
        }
      ]
    },
    {
      "name": "luajit",
      "bom-ref": "lib:luajit",
      "type": "library",
      "version": "2.1.0-0.16beta3.el8.x86_64",
      "supplier": {
        "name": "Fedora Project"
      },
      "licenses": [
        {
          "license": {
            "id": "MIT"
          }
        }
      ]
    },
    {
      "name": "net-snmp-libs",
      "bom-ref": "lib:net-snmp-libs",
      "type": "library",
      "version": "1:5.8-30.0.1.el8.x86_64",
      "supplier": {
        "name": "Oracle America"
      },
      "licenses": [
        {
          "license": {
            "id": "BSD-3-Clause"
          }
        }
      ]
    },
    {
      "name": "net-snmp-agent-libs",
      "bom-ref": "lib:net-snmp-agent-libs",
      "type": "library",
      "version": "1:5.8-30.0.1.el8.x86_64",
      "supplier": {
        "name": "Oracle America"
      },
      "licenses": [
        {
          "license": {
            "id": "BSD-3-Clause"
          }
        }
      ]
    },
    {
      "name": "libnghttp2",
      "bom-ref": "lib:libnghttp2",
      "type": "library",
      "version": "1.33.0-6.el8_10.1.x86_64",
      "supplier": {
        "name": "Oracle America"
      },
      "licenses": [
        {
          "license": {
            "id": "MIT"
          }
        }
      ]
    },
    {
      "name": "re2",
      "bom-ref": "lib:re2",
      "type": "library",
      "version": "20190801-17.el8.x86_64",
      "supplier": {
        "name": "Fedora Project"
      },
      "licenses": [
        {
          "license": {
            "name": "BSD-3-Clause"
          }
        }
      ]
    },
    {
      "name": "libsodium",
      "bom-ref": "lib:libsodium",
      "type": "library",
      "version": "1.0.18-2.el8.x86_64",
      "supplier": {
        "name": "Fedora Project"
      },
      "licenses": [
        {
          "license": {
            "name": "ISC"
          }
        }
      ]
    },
    {
      "name": "libstdc++",
      "bom-ref": "lib:libstdc++",
      "type": "library",
      "version": "8.5.0-22.0.1.el8_10.x86_64",
      "supplier": {
        "name": "Oracle America"
      },
      "licenses": [
        {
          "license": {
            "name": "GPLv3+ and GPLv3+ with exceptions and GPLv2+ with exceptions and LGPLv2+ and BSD"
          }
        }
      ]
    },
    {
      "name": "systemd-libs",
      "bom-ref": "lib:systemd-libs",
      "type": "library",
      "version": "239-82.0.2.el8.x86_64",
      "supplier": {
        "name": "Oracle America"
      },
      "licenses": [
        {
          "license": {
            "name": "LGPLv2+ and MIT"
          }
        }
      ]
    },
    {
      "name": "libxdp",
      "bom-ref": "lib:libxdp",
      "type": "library",
      "version": "1.2.10-1.el8.x86_64",
      "supplier": {
        "name": "Oracle America"
      },
      "licenses": [
        {
          "license": {
            "id": "GPL-2.0-only"
          }
        }
      ]
    },
    {
      "name": "rust",
      "bom-ref": "lib:rust",
      "type": "library",
      "version": "1.78.0",
      "publisher": "https://www.rust-lang.org/",
      "licenses": [
        {
          "license": {
            "id": "MIT"
          }
        }
      ],
      "hashes": [
        {
          "alg": "SHA-256",
          "content": "1377999f189d328ec183676c0e69f21af16e450f8d67f10d1f6cf746951a1d64"
        }
      ]
    },
    {
      "name": "quiche",
      "bom-ref": "lib:quiche",
      "type": "library",
      "version": "0.22.0",
      "publisher": "https://github.com/cloudflare/quiche",
      "licenses": [
        {
          "license": {
            "name": "BSD-2-Clause"
          }
        }
      ],
      "hashes": [
        {
          "alg": "SHA-256",
          "content": "0af8744b07038ee4af8cdb94dd4c11f1a730001944a0ef2f3f03e63715b15268"
        }
      ]
    },
    {
      "name": "h2o",
      "bom-ref": "lib:h2o",
      "type": "library",
      "version": "2.2.6-pdns2",
      "publisher": "https://github.com/h2o/h2o",
      "licenses": [
        {
          "license": {
            "id": "MIT"
          }
        }
      ],
      "hashes": [
        {
          "alg": "SHA-256",
          "content": "e25959c3f9a102e7a332ca0bb8b3f533eb14919d5a60ca999730c2ebee4b548f"
        }
      ]
    }
  ],
  "dependencies": [
    {
      "ref": "pkg:dnsdist",
      "dependsOn": [
        "lib:shadow-utils",
        "lib:systemd",
        "lib:libbpf",
        "lib:glibc",
        "lib:libcap",
        "lib:tinycdb",
        "lib:openssl-libs",
        "lib:libedit",
        "lib:fstrm",
        "lib:libgcc",
        "lib:gnutls",
        "lib:lmdb-libs",
        "lib:luajit",
        "lib:net-snmp-libs",
        "lib:net-snmp-agent-libs",
        "lib:libnghttp2",
        "lib:re2",
        "lib:libsodium",
        "lib:libstdc++",
        "lib:systemd-libs",
        "lib:libxdp",
        "lib:rust",
        "lib:quiche",
        "lib:h2o"
      ]
    }
  ]
}
```

</details>

File: builder-support/helpers/generate-sbom-dnf.py [hunk 1]
```diff
@@ -44,10 +44,13 @@ def getPackageInformations(pkgDB, packageName):
 def addDependencyToSBOM(sbom, appInfos, pkg):
     bomRef = 'lib:' + pkg.name
     component = { 'name': pkg.name, 'bom-ref': bomRef, 'type': 'library'}
+    print(component)
```
Comment (rgacogne) [inline builder-support/helpers/generate-sbom-dnf.py:47]: I'm guessing this should go now? :)

**Review by rgacogne — APPROVED** [review]
Output looks great, nice work!

File: builder-support/helpers/generate-sbom-dnf.py (hunk 1 above)
Comment (Habbie) [reply builder-support/helpers/generate-sbom-dnf.py:47]: yes!

Comment (Habbie) [reply builder-support/helpers/generate-sbom-dnf.py]: amended

**Pull request merged**